request-manager/server/server.go

```diff
@@ -26,6 +30,45 @@ func NewServer(appCtx app.Context) *Server {
 	}
 }
 
+// Run runs the Request Manager API in the foreground. It returns when the API stops.
+func (s *Server) Run() error {
+	if s.api == nil {
+		panic("Server.Run called before Server.Boot")
+	}
+
+	go s.waitForShutdown()
```

Comment:
Wait for shutdown after starting? Else, in pathological case, we could shutdown before we've started.

Reply:
Sure, I can move this to after starting the resumer. It doesn't really matter - it just closes shutdownChan, which would stop the resumer right away after it started.